src/fontcache.cpp

```diff
@@ -1056,9 +1056,301 @@ static void LoadWin32Font(FontSize fs)
+
+	if (bounds.size.width > 0) {
+		/* Glyph is not a white-space glyph. Render it to a bitmap context. */
+
+		/* We only need the alpha channel, as we apply our own colour constants to the sprite. */
+		int pitch = Align(bb_width, 16);
+		byte *bmp = CallocT<byte>(bb_height * pitch);
+		CFAutoRelease<CGContextRef> context(CGBitmapContextCreate(bmp, bb_width, bb_height, 8, pitch, nullptr, kCGImageAlphaOnly));
+		/* Set antialias according to requirements. */
+		CGContextSetAllowsAntialiasing(context.get(), aa);
+		CGContextSetAllowsFontSubpixelPositioning(context.get(), aa);
+		CGContextSetAllowsFontSubpixelQuantization(context.get(), !aa);
+		CGContextSetShouldSmoothFonts(context.get(), false);
+
+		float offset = aa ? 0.0f : 0.5f; // Align to pixel grid for non-AA.
+		CGPoint pos{offset - bounds.origin.x, offset - bounds.origin.y};
+		CTFontDrawGlyphs(this->font.get(), &glyph, &pos, 1, context.get());
+
+		/* Draw shadow for medium size. */
+		if (this->fs == FS_NORMAL && !aa) {
+			for (uint y = 0; y < bb_height; y++) {
+				for (uint x = 0; x < bb_width; x++) {
+					if (bmp[y * pitch + x] > 0) {
+						sprite.data[1 + x + (1 + y) * sprite.width].m = SHADOW_COLOUR;
+						sprite.data[1 + x + (1 + y) * sprite.width].a = aa ? bmp[x + y * pitch] : 0xFF;
+					}
+				}
+			}
+		}
+
+		/* Extract pixel data. */
+		for (uint y = 0; y < bb_height; y++) {
+			for (uint x = 0; x < bb_width; x++) {
+				if (bmp[y * pitch + x] > 0) {
+					sprite.data[x + y * sprite.width].m = FACE_COLOUR;
+					sprite.data[x + y * sprite.width].a = aa ? bmp[x + y * pitch] : 0xFF;
+				}
+			}
+		}
+	}
+
+	GlyphEntry new_glyph;
+	new_glyph.sprite = BlitterFactory::GetCurrentBlitter()->Encode(&sprite, AllocateFont);
+	new_glyph.width = (byte)std::round(CTFontGetAdvancesForGlyphs(this->font.get(), kCTFontOrientationDefault, &glyph, nullptr, 1));
+	this->SetGlyphPtr(key, &new_glyph);
+
+	return new_glyph.sprite;
+}
+
+/**
+ * Loads the TrueType font.
+ * If a CoreText font description is present, e.g. from the automatic font
+ * fallback search, use it. Otherwise, try to resolve it by font name.
+ * @param fs The font size to load.
+ */
+static void LoadCoreTextFont(FontSize fs)
+{
+	static const char *SIZE_TO_NAME[] = { "medium", "small", "large", "mono" };
+
+	FreeTypeSubSetting *settings = nullptr;
+	switch (fs) {
+		default: NOT_REACHED();
+		case FS_SMALL:  settings = &_freetype.small;  break;
+		case FS_NORMAL: settings = &_freetype.medium; break;
+		case FS_LARGE:  settings = &_freetype.large;  break;
+		case FS_MONO:   settings = &_freetype.mono;   break;
+	}
+
+	if (StrEmpty(settings->font)) return;
+
+	CFAutoRelease<CTFontDescriptorRef> font_ref;
+
+	if (settings->os_handle != nullptr) {
+		font_ref.reset(static_cast<CTFontDescriptorRef>(const_cast<void *>(settings->os_handle)));
+		CFRetain(font_ref.get()); // Increase ref count to match a later release.
+	}
+
+	if (!font_ref && MacOSVersionIsAtLeast(10, 6, 0)) {
```

Comment:
Is this 10.6+ check needed, given we are 10.9+?

Reply:
People could still compile OpenTTD themselves on an older version, though 10.6 is rather old now admittedly. If it's new code, it's probably fine not to include such a check.

Reply:
My personal opinion here is that as long as backwards compatibility is simple (i.e. a single if like here), I try account for it.